src/extension/ai/converters.ts

```diff
@@ -0,0 +1,86 @@
+// Conversion routines between the protocol buffer format and VSCode's representation of a notebook
+//
+// See ../vscode_apis.md for an exlanation. It is very helpful for understanding this folder.
+
+import * as vscode from 'vscode'
+//import * as metadata from './metadata'
+//import * as constants from './constants'
+import {
+  Notebook,
+  Cell,
+  CellKind,
+  CellOutput,
+  CellOutputItem,
+} from '@buf/stateful_runme.community_timostamm-protobuf-ts/runme/parser/v1/parser_pb'
+
+// cellToCellData converts a NotebookCell to a NotebookCellData.
+// NotebookCell is an interface used by the editor.
+// NotebookCellData is a concrete class.
+export function cellToCellData(cell: vscode.NotebookCell): vscode.NotebookCellData {
+  let data = new vscode.NotebookCellData(
+    cell.kind,
+    cell.document.getText(),
+    cell.document.languageId,
+  )
+
+  data.metadata = cell.metadata
+  data.outputs = []
+  for (let o of cell.outputs) {
+    data.outputs.push(o)
+  }
+  return data
+}
+
+// cellDataToProto converts an instance of NotebookCellData to the proto.
+export function cellDataToProto(cell: vscode.NotebookCellData): Cell {
+  let cellPb = Cell.create()
+  cellPb.value = cell.value
+  cellPb.languageId = cell.languageId
+  if (cell.kind === vscode.NotebookCellKind.Code) {
+    cellPb.kind = CellKind.CODE
+  } else {
+    cellPb.kind = CellKind.MARKUP
+  }
+
+  // TODO(jeremy):  cell.metadata is type map[string]Any; cellPb.metadata is type map[string]string
```

Comment:
@sourishkrout Is there existing conversion code I can use here? If not is there a way to do this other than just testing if the values are strings and if not ignoring them?

Reply:
I believe this static method is what you're looking for:

https://github.com/stateful/vscode-runme/blob/main/src/extension/serializer.ts#L740-L757